src/debug.lisp

```diff
@@ -173,9 +173,24 @@
   "Lookup the type of value SYMBOL in the global environment"
   (tc:lookup-value-type entry:*global-environment* symbol))
 
+(defun coalton:describe-type-of (symbol)
+  "Print the type of value SYMBOL along with its type aliases and return it"
+  (let ((tc:*pprint-type-aliases* t)
+        (type (tc:lookup-value-type entry:*global-environment* symbol)))
```

Comment:
It looks like these are inconsistent with how it prints aliases that are imported from other packages.

If you :use a package then this will print the alias without the package. If you don't :use a package, but do something like this:
```lisp
  (declare adds (aliases2:Int -> Integer -> Integer))
```
then you will get an error message that does indeed print the package:
```
  --> <macroexpansion>:1:15
   |
 1 |  (COALTON (ADDS "hi" 2))
   |                 ^^^^ Expected type '[ALIASES2:INT := INTEGER]' but got 'STRING'
```

I think, since this is for debugging purposes, it would be helpful to always print the package that a type is from unless it is from the current package or coalton-prelude. Maybe any of the library functions.

---
Here is another example. In this error message, we don't `:use` the package. In the error message, the alias is printed with one : but the type it refers to (from the same package) is printed with two ::'s:

"aliases2"
```lisp
(in-package :cl-user)
(defpackage :aliases2
  (:use
   #:coalton
   #:coalton-prelude)
  (:export
   :Int
   :Ham
   :Swiss
   :Cheese))
(in-package :aliases2)

(coalton-toplevel
  (define-type MoldyMilk Ham Swiss)

  (define-type-alias Cheese MoldyMilk))

(coalton-toplevel
  (define-type-alias Int Integer))
```

aliases (main package)
```lisp
(in-package :cl-user)
(defpackage :aliases
  (:use
   #:coalton
   #:coalton-prelude))
   ;; #:aliases2)) <---- not using the other package
(in-package :aliases)

(coalton-toplevel
  (declare value-of-cheese (aliases2:Cheese -> Integer))
  (define (value-of-cheese chs)
    (match chs
      ((aliases2:Ham) 10)
      ((aliases2:Swiss) 20))))

(coalton (value-of-cheese "not cheese"))
```


```
error: Type mismatch
  --> <macroexpansion>:1:26
   |
 1 |  (COALTON (VALUE-OF-CHEESE "not cheese"))
   |                            ^^^^^^^^^^^^ Expected type '[ALIASES2:CHEESE := ALIASES2::MOLDYMILK]' but got 'STRING'
```

Reply:
I will take a closer look at the behavior of `describe-type-of`. But regarding your examples, I believe those are all expected behaviors.

First, the use of `:` versus `::` depends on whether is a symbol is exported. If a package `p` exports a symbol `s`, and you do not want to `:use` package `p`, then you can use the symbol with `p:s`. If `p` does not export `s`, then you can still use it with `p::s`. In your example, `aliases2` exports `cheese` but not `moldymilk`.

Second, when package `p` `:uses` package `q`, the symbols from `q` receive the same treatment locally, in `p`, as any other symbol defined in `p`, so my personal opinion is that it is not inconsistent for those types aliases to be printed without the `q:` prefix.

Reply:
That makes sense 👍